### ISSUE TITLE
SlevomatCodingStandard.Namespaces.UseSpacing not corresponding to config

### ISSUE THREAD

**Issue by oojacoboo** [slevomat/coding-standard#662]
```
"squizlabs/php_codesniffer": "~3.4",
"slevomat/coding-standard": "~5.0",
```

I'm getting the following error:

>Expected 1 lines after last use statement, found 2.
>(SlevomatCodingStandard.Namespaces.UseSpacing.IncorrectLinesCountAfterLastUse)

With this config...

```xml
<?xml version="1.0" encoding="UTF-8"?>

<ruleset name="Foo" namespace="Foo\CodingStandard">
    <description>Foo default coding standard</description>
    <ini name="memory_limit" value="1024M" />

    <!--
        IMPORTANT: all exclude patterns for files and paths should be added to the project in a
        ruleset file and passed to the phpcs CLI command.
    -->

    <exclude-pattern>*.(css|js|json|html|yml|yaml)</exclude-pattern>

    <arg name="colors" />
    <arg value="s" /> <!-- command line argument "-s", now output will include the rule identifier code -->


    <!-- Including custom rules from vendor libs b/c they add awesome stuff -->
    <!-- Note: these are relative to the vendor dir in projects where they are pulled, consider fixing -->
    <!-- Note: we have to exclude some rules that this ruleset includes directly in addition to the actual rulesets below -->
	<rule ref="./vendor/slevomat/coding-standard/SlevomatCodingStandard/ruleset.xml">
        <exclude name="Consistence.Exceptions.ExceptionDeclaration.IncorrectExceptionDirectory" />

        <exclude name="SlevomatCodingStandard.Classes.SuperfluousAbstractClassNaming.SuperfluousPrefix" />
        <exclude name="SlevomatCodingStandard.Classes.SuperfluousExceptionNaming.SuperfluousSuffix" />
        <exclude name="SlevomatCodingStandard.Classes.SuperfluousInterfaceNaming.SuperfluousSuffix" />
        <exclude name="SlevomatCodingStandard.Commenting.DisallowOneLinePropertyDocComment.OneLinePropertyComment" />
        <exclude name="SlevomatCodingStandard.Commenting.UselessInheritDocComment.UselessInheritDocComment" />
        <exclude name="SlevomatCodingStandard.ControlStructures.ControlStructureSpacing.IncorrectLinesCountBeforeControlStructure" />
        <exclude name="SlevomatCodingStandard.ControlStructures.DisallowEmpty.DisallowedEmpty" />
        <exclude name="SlevomatCodingStandard.ControlStructures.DisallowShortTernaryOperator.DisallowedShortTernaryOperator" />
        <exclude name="SlevomatCodingStandard.ControlStructures.NewWithoutParentheses.UselessParentheses" />
        <exclude name="SlevomatCodingStandard.ControlStructures.RequireMultiLineTernaryOperator.MultiLineTernaryOperatorNotUsed" />
        <exclude name="SlevomatCodingStandard.ControlStructures.RequireTernaryOperator.TernaryOperatorNotUsed" />
        <exclude name="SlevomatCodingStandard.ControlStructures.RequireYodaComparison.RequiredYodaComparison" />
        <exclude name="SlevomatCodingStandard.Files.TypeNameMatchesFileName" />
        <exclude name="SlevomatCodingStandard.Namespaces.FullyQualifiedClassNameAfterKeyword.NonFullyQualifiedExtends" />
        <exclude name="SlevomatCodingStandard.Namespaces.FullyQualifiedClassNameAfterKeyword.NonFullyQualifiedUse" />
        <exclude name="SlevomatCodingStandard.Namespaces.FullyQualifiedClassNameAfterKeyword.NonFullyQualifiedImplements" />
        <exclude name="SlevomatCodingStandard.Namespaces.FullyQualifiedClassNameInAnnotation.NonFullyQualifiedClassName" />
        <exclude name="SlevomatCodingStandard.Namespaces.FullyQualifiedExceptions.NonFullyQualifiedException" />
        <exclude name="SlevomatCodingStandard.Namespaces.FullyQualifiedGlobalFunctions.NonFullyQualified" />
        <exclude name="SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly.PartialUse" />
        <exclude name="SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly.ReferenceViaFullyQualifiedName" />
        <exclude name="SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly.ReferenceViaFullyQualifiedNameWithoutNamespace" />
        <exclude name="SlevomatCodingStandard.Namespaces.UseOnlyWhitelistedNamespaces.NonFullyQualified" />
        <exclude name="SlevomatCodingStandard.Operators.DisallowIncrementAndDecrementOperators.DisallowedPostDecrementOperator" />
        <exclude name="SlevomatCodingStandard.Operators.DisallowIncrementAndDecrementOperators.DisallowedPostIncrementOperator" />
        <exclude name="SlevomatCodingStandard.Operators.DisallowIncrementAndDecrementOperators.DisallowedPreDecrementOperator" />
        <exclude name="SlevomatCodingStandard.Operators.DisallowIncrementAndDecrementOperators.DisallowedPreIncrementOperator" />
        <exclude name="SlevomatCodingStandard.Types.EmptyLinesAroundTypeBraces.NoEmptyLineBeforeClosingBrace" />
        <exclude name="SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingTraversableParameterTypeHintSpecification" />
        <exclude name="SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingTraversablePropertyTypeHintSpecification" />
        <exclude name="SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingTraversableReturnTypeHintSpecification" />
        <exclude name="SlevomatCodingStandard.TypeHints.TypeHintDeclaration.UselessDocComment" />

        <exclude name="Squiz.Commenting.FunctionComment.SpacingAfterParamName" />
    </rule>
    <!-- Slevomat rule overrides -->
    <rule ref="SlevomatCodingStandard.Classes.TraitUseSpacing">
        <properties>
            <property name="linesCountAfterLastUseWhenLastInClass" value="2" />
            <property name="linesCountAfterLastUse" value="2" />
        </properties>
    </rule>
    <rule ref="SlevomatCodingStandard.Commenting.DocCommentSpacing">
        <properties>
            <property name="linesCountBeforeFirstContent" value="0" />
            <property name="linesCountBetweenDescriptionAndAnnotations" value="1" />
            <property name="linesCountBetweenDifferentAnnotationsTypes" value="1" />
            <property name="linesCountAfterLastContent" value="0" />
        </properties>
    </rule>
    <rule ref="SlevomatCodingStandard.Files.TypeNameMatchesFileName">
        <properties>
            <property name="skipDirs" type="array" value="lib" />
            <property name="skipDirs" type="array" value="plugins" />
            <property name="skipDirs" type="array" value="public" />
            <property name="skipDirs" type="array" value="test" />
            <property name="skipDirs" type="array" value="vendors" />
        </properties>
    </rule>
    <rule name="SlevomatCodingStandard.Namespaces.UseOnlyWhitelistedNamespaces">
        <properties>
            <property name="namespacesRequiredToUse" value="RentPost\Cog\Exception" />
        </properties>
    </rule>
    <rule name="SlevomatCodingStandard.Namespaces.UseSpacing">
        <properties>
            <property name="linesCountBeforeFirstUse" value="1" />
            <property name="linesCountBetweenUseTypes" value="1" />
            <property name="linesCountAfterLastUse" value="2" />
        </properties>
    </rule>
    <rule ref="SlevomatCodingStandard.TypeHints.DeclareStrictTypes">
        <properties>
            <property name="newlinesCountBetweenOpenTagAndDeclare" value="2" /> <!-- It counts the same line as the <?php statment -->
            <property name="spacesCountAroundEqualsSign" value="1" />
        </properties>
    </rule>


    <!-- Generic Rules -->
    <rule ref="Generic.Classes.DuplicateClassName" />
    <rule ref="Generic.CodeAnalysis.ForLoopShouldBeWhileLoop" />
    <rule ref="Generic.CodeAnalysis.ForLoopWithTestFunctionCall" />
    <rule ref="Generic.CodeAnalysis.JumbledIncrementer" />
    <rule ref="Generic.CodeAnalysis.UnconditionalIfStatement" />
    <rule ref="Generic.CodeAnalysis.UnnecessaryFinalModifier" />
    <rule ref="Generic.CodeAnalysis.UnusedFunctionParameter" />
    <rule ref="Generic.CodeAnalysis.UselessOverridingMethod" />
    <rule ref="Generic.Commenting.DocComment">
		<!-- Having a @see or @internal tag before the @param tags is fine. -->
		<exclude name="Generic.Commenting.DocComment.ParamNotFirst" />
        <exclude name="Generic.Commenting.DocComment.ContentAfterOpen" />
        <exclude name="Generic.Commenting.DocComment.ContentBeforeClose" />
        <!-- Added this until there is a PSR5 rule that supports docblock style single line comments for variables and properties -->
        <exclude name="Generic.Commenting.DocComment.MissingShort" />
        <exclude name="Generic.Commenting.DocComment.SpacingBeforeTags" />
        <!-- Other -->
        <exclude name="Generic.Commenting.DocComment.TagValueIndent" />
	</rule>
    <rule ref="Generic.Commenting.Todo" />
    <rule ref="Generic.ControlStructures.InlineControlStructure" />
    <rule ref="Generic.Files.ByteOrderMark" />
    <rule ref="Generic.Files.LineEndings" />
    <rule ref="Generic.Files.LineLength">
        <properties>
            <property name="lineLimit" value="100" />
            <!-- using fairly higher limits just to keep some old code happy -->
            <property name="absoluteLineLimit" value="180" />
        </properties>
    </rule>
    <rule ref="Generic.Formatting.DisallowMultipleStatements" />
    <rule ref="Generic.Formatting.NoSpaceAfterCast" />
    <rule ref="Generic.Formatting.SpaceAfterCast">
        <exclude name="Generic.Formatting.SpaceAfterCast.NoSpace" />
    </rule>
    <rule ref="Generic.Functions.CallTimePassByReference" />
    <rule ref="Generic.Functions.FunctionCallArgumentSpacing" />
    <rule ref="Generic.Functions.OpeningFunctionBraceBsdAllman">
        <exclude name="Generic.Functions.OpeningFunctionBraceBsdAllman.BraceOnSameLine" />
    </rule>
    <rule ref="Generic.Metrics.CyclomaticComplexity" />
    <rule ref="Generic.Metrics.NestingLevel">
        <properties>
            <property name="nestingLevel" value="3" />
            <property name="absoluteNestingLevel" value="8" />
        </properties>
    </rule>
    <rule ref="Generic.NamingConventions.ConstructorName" />
    <rule ref="Generic.NamingConventions.UpperCaseConstantName" />
    <rule ref="Generic.NamingConventions.CamelCapsFunctionName">
        <exclude name="Generic.NamingConventions.CamelCapsFunctionName.ScopeNotCamelCaps" />
    </rule>
    <rule ref="Generic.PHP.DeprecatedFunctions" />
    <rule ref="Generic.PHP.DisallowShortOpenTag" />
    <rule ref="Generic.PHP.ForbiddenFunctions" />
    <rule ref="Generic.PHP.LowerCaseConstant" />
    <rule ref="Generic.PHP.NoSilencedErrors" />
    <rule ref="Generic.Strings.UnnecessaryStringConcat" />
    <rule ref="Generic.WhiteSpace.DisallowSpaceIndent">
        <exclude name="Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed" />
    </rule>
    <rule ref="Generic.WhiteSpace.DisallowTabIndent" />
    <rule ref="Generic.WhiteSpace.ScopeIndent" />


    <!-- MySource Rules -->
    <rule ref="MySource.PHP.EvalObjectFactory" />
    <rule ref="MySource.PHP.ReturnFunctionValue">
        <exclude name="MySource.PHP.ReturnFunctionValue.NotAssigned" />
    </rule>
    <rule ref="MySource.PHP.AjaxNullComparison" />


    <!-- PEAR Convention Rules -->
    <rule ref="PEAR.Classes.ClassDeclaration" />
    <rule ref="PEAR.Commenting">
        <!-- Don't require normal file comments -->
        <exclude name="PEAR.Commenting.FileComment" />
        <!-- Be less strict about class comments -->
        <exclude name="PEAR.Commenting.ClassComment.MissingLicenseTag" />
		<exclude name="PEAR.Commenting.ClassComment.MissingCategoryTag" />
		<exclude name="PEAR.Commenting.ClassComment.MissingLinkTag" />
		<exclude name="PEAR.Commenting.ClassComment.MissingPackageTag" />
        <exclude name="PEAR.Commenting.FunctionComment.MissingReturn" />
        <exclude name="PEAR.Commenting.FunctionComment.SpacingAfterParamName" />
        <exclude name="PEAR.Commenting.FunctionComment.SpacingAfterParamType" />
        <!-- Parameters names should be descriptive by nature, a comment isn't always required -->
        <exclude name="PEAR.Commenting.FunctionComment.MissingParamComment" />
    </rule>
    <rule ref="PEAR.Commenting.FunctionComment">
    </rule>
    <rule ref="PEAR.Commenting.InlineComment" />
    <rule ref="PEAR.ControlStructures.ControlSignature" />
    <rule ref="PEAR.ControlStructures.MultiLineCondition">
        <exclude name="PEAR.ControlStructures.MultiLineCondition.CloseBracketNewLine" />
    </rule>
    <rule ref="PEAR.Files.IncludingFile" />
    <rule ref="PEAR.Formatting.MultiLineAssignment" />
    <rule ref="PEAR.Functions.FunctionCallSignature">
        <exclude name="PEAR.Functions.FunctionCallSignature.CloseBracketLine" />
        <exclude name="PEAR.Functions.FunctionCallSignature.ContentAfterOpenBracket" />
    </rule>
    <rule ref="PEAR.Functions.FunctionDeclaration">
        <exclude name="PEAR.Functions.FunctionDeclaration.NewlineBeforeOpenBrace" />
    </rule>
    <rule ref="PEAR.Functions.ValidDefaultValue" />
    <rule ref="PEAR.NamingConventions.ValidClassName" />
    <rule ref="PEAR.WhiteSpace.ObjectOperatorIndent">
        <exclude name="PEAR.WhiteSpace.ObjectOperatorIndent.Incorrect" />
    </rule>
    <rule ref="PEAR.WhiteSpace.ScopeClosingBrace" />
    <rule ref="PEAR.WhiteSpace.ScopeIndent">
        <exclude name="PEAR.WhiteSpace.ScopeIndent.IncorrectExact" />
    </rule>


    <!-- PSR Standard Rules -->
    <rule ref="PSR2">
        <exclude name="PSR2.Namespaces.UseDeclaration.SpaceAfterLastUse" />
    </rule>


    <!-- Squiz Rules -->
    <rule ref="Squiz.Arrays.ArrayDeclaration">
		<!-- Ignoring the Squiz indentation rules as normalized arrays are preferred.
		     Unfortunately there is currently no upstream sniff to check for this. -->
        <exclude name="Squiz.Arrays.ArrayDeclaration.DoubleArrowNotAligned" />
		<exclude name="Squiz.Arrays.ArrayDeclaration.KeyNotAligned" />
		<exclude name="Squiz.Arrays.ArrayDeclaration.ValueNotAligned" />
		<exclude name="Squiz.Arrays.ArrayDeclaration.CloseBraceNotAligned" />

		<!-- Single and multi-line arrays are both allowed. -->
		<exclude name="Squiz.Arrays.ArrayDeclaration.SingleLineNotAllowed" />
		<exclude name="Squiz.Arrays.ArrayDeclaration.MultiLineNotAllowed" />
	</rule>
    <rule ref="Squiz.Commenting">
        <exclude name="Squiz.Commenting.BlockComment.HasEmptyLineBefore" />
        <exclude name="Squiz.Commenting.BlockComment.WrongStart" />
        <exclude name="Squiz.Commenting.ClassComment.TagNotAllowed" />
        <exclude name="Squiz.Commenting.ClosingDeclarationComment.Missing" />
        <exclude name="Squiz.Commenting.FileComment.Missing" />
        <exclude name="Squiz.Commenting.FileComment.MissingPackageTag" />
        <exclude name="Squiz.Commenting.FunctionComment.EmptyThrows" />
        <exclude name="Squiz.Commenting.FunctionComment.IncorrectParamVarName" />
        <exclude name="Squiz.Commenting.FunctionComment.InvalidNoReturn" />
        <exclude name="Squiz.Commenting.FunctionComment.MissingParamComment" />
        <exclude name="Squiz.Commenting.FunctionComment.MissingReturn" />
        <exclude name="Squiz.Commenting.FunctionComment.ParamCommentFullStop" />
        <exclude name="Squiz.Commenting.FunctionComment.SpacingAfterParamType" />
        <exclude name="Squiz.Commenting.FunctionCommentThrowTag.Missing" />
        <exclude name="Squiz.Commenting.InlineComment.InvalidEndChar" />
        <exclude name="Squiz.Commenting.InlineComment.DocBlock" />
        <exclude name="Squiz.Commenting.LongConditionClosingComment.Missing" />
        <exclude name="Squiz.Commenting.PostStatementComment.Found" />
        <exclude name="Squiz.Commenting.VariableComment.IncorrectVarType" />
        <exclude name="Squiz.Commenting.VariableComment.TagNotAllowed" />
    </rule>
    <rule ref="Squiz.Functions.MultiLineFunctionDeclaration">
        <exclude name="Squiz.Functions.MultiLineFunctionDeclaration.NewlineBeforeOpenBrace" /> <!-- clashes with OpeningFunctionBraceBsdAllman -->
    </rule>
    <rule ref="Squiz.PHP.CommentedOutCode">
        <properties>
            <property name="maxPercentage" value="25" />
        </properties>
    </rule>
    <rule ref="Squiz.PHP.DisallowComparisonAssignment">
        <exclude name="Squiz.PHP.DisallowComparisonAssignment.AssignedBool" />
        <exclude name="Squiz.PHP.DisallowComparisonAssignment.AssignedComparison" />
    </rule>
    <rule ref="Squiz.PHP.DisallowMultipleAssignments" />
    <rule ref="Squiz.PHP.DisallowSizeFunctionsInLoops" />
    <rule ref="Squiz.PHP.DiscouragedFunctions">
        <properties>
            <property name="error" value="true" />
        </properties>
    </rule>
    <rule ref="Squiz.PHP.Eval" />
    <rule ref="Squiz.PHP.GlobalKeyword">
        <exclude name="Squiz.PHP.GlobalKeyword.NotAllowed" />
    </rule>
    <rule ref="Squiz.PHP.Heredoc" />
    <rule ref="Squiz.PHP.InnerFunctions" />
    <rule ref="Squiz.PHP.LowercasePHPFunctions" />
    <rule ref="Squiz.PHP.NonExecutableCode" />
    <rule ref="Squiz.Scope.MemberVarScope" />
    <rule ref="Squiz.Scope.MethodScope" />
    <rule ref="Squiz.Scope.StaticThisUsage" />
    <rule ref="Squiz.Strings" />
    <rule ref="Squiz.Strings.DoubleQuoteUsage">
        <exclude name="Squiz.Strings.DoubleQuoteUsage.ContainsVar" />
    </rule>
    <rule ref="Squiz.Strings.ConcatenationSpacing">
        <properties>
            <property name="spacing" value="1" />
            <property name="ignoreNewlines" value="true" />
        </properties>
    </rule>
    <rule ref="Squiz.WhiteSpace.CastSpacing" />
    <rule ref="Squiz.WhiteSpace.ControlStructureSpacing" />
    <rule ref="Squiz.WhiteSpace.FunctionOpeningBraceSpace" />
    <rule ref="Squiz.WhiteSpace.FunctionSpacing">
        <properties>
            <property name="spacing" value="2" />
            <property name="spacingBeforeFirst" value="2" />
            <property name="spacingAfterLast" value="0" />
        </properties>
        <exclude name="Squiz.WhiteSpace.FunctionSpacing.Before" />
        <exclude name="Squiz.WhiteSpace.FunctionSpacing.BeforeFirst" />
    </rule>
    <rule ref="Squiz.WhiteSpace.LanguageConstructSpacing" />
    <rule ref="Squiz.WhiteSpace.LogicalOperatorSpacing" />
    <rule ref="Squiz.WhiteSpace.MemberVarSpacing">
        <exclude name="Squiz.WhiteSpace.MemberVarSpacing.FirstIncorrect" />
        <properties>
            <property name="spacingBeforeFirst" value="2" /> <!-- not working yet, waiting on next version -->
            <property name="spacing" value="1" />
        </properties>
    </rule>
    <rule ref="Squiz.WhiteSpace.ObjectOperatorSpacing">
        <properties>
            <property name="ignoreNewlines" value="true" />
        </properties>
    </rule>
    <rule ref="Squiz.WhiteSpace.OperatorSpacing">
        <properties>
            <property name="ignoreNewlines" value="true" />
        </properties>
    </rule>
    <rule ref="Squiz.WhiteSpace.PropertyLabelSpacing" />
    <rule ref="Squiz.WhiteSpace.ScopeClosingBrace" />
    <rule ref="Squiz.WhiteSpace.ScopeKeywordSpacing" />
    <rule ref="Squiz.WhiteSpace.SemicolonSpacing" />
    <rule ref="Squiz.WhiteSpace.SuperfluousWhitespace">
        <properties>
            <property name="ignoreBlankLines" value="true" />
        </properties>
    </rule>


    <!-- Zend Rules -->
    <rule ref="Zend.Files.ClosingTag" />
    <rule ref="Zend.NamingConventions.ValidVariableName">
        <exclude name="Zend.NamingConventions.ValidVariableName.PrivateNoUnderscore" />
    </rule>

</ruleset>
```

Any ideas what's going on here?  I've tried tweaking the order of the rules and haven't had any luck.

**Comment by kukulich** [comment]
@oojacoboo I don’t see any code...

**Comment by oojacoboo** [comment]
@kukulich I figured it was pretty self-explanatory.  The error isn't inaccurate in it's description of the code.  Here is an example.

```
<?php

namespace Foo;

use Bar\Baz;
use Whatever\Thing;


/**
 * Docblock here
 */
class A
{
}
```

**Comment by kukulich** [comment]
I don't see any noticeable problem. Try to remove `<rule ref="./vendor/slevomat/coding-standard/SlevomatCodingStandard/ruleset.xml">` and include every sniff separately.

https://github.com/slevomat/coding-standard#choose-which-sniffs-to-run
